### DISCUSSION TITLE
RFC 0005: move to Acceptance (Example overlay)

### DISCUSSION THREAD

**Discussion by carlyh-micb** [the-human-colossus-foundation/oca-spec#60] [General]
I would like to propose that, similar to the framing overlay, the example overlay be accepted into the specification.

What is the process for approving overlays into the specification?

https://github.com/the-human-colossus-foundation/oca-spec/blob/master/RFCs/0005-ExampleOverlay.md

**Comment by blelump** [comment]
OCA is a coherent ecosystem. Inputs cannot be entered randomly because OCA provides the proper context to ensure data entry matches the constraints (attribute type, etc.). This is the validation process. Therefore, mechanically, what is expected is always evident, and no example is required. For the edge cases where such a feature could be helpful, it becomes the role of `community overlay` to bring such a feature where the community maintains it. 

If you can think of a case where capture base and overlays are not enough to provide such `proper context,` there's clearly a lack of mechanics we need to improve.

**Comment by carlyh-micb** [comment]
While the specification may provide all the information, an example is always helpful. The OCA specification even recognizes this and gives examples right in the spec. 
Providing examples help users grasp the expected format or content of the data they need to input. Reading the rules is often not sufficient for human users (while being sufficient for machines) and users experience less mental load and perform better when they can compare the rules to an example.
Your suggestion of a community overlay is what we have been discussing as an extension?

**Reply by blelump** [reply]
Let's consider a data model built upon a relational database. Does it come with an example of how to use it? Nope. It eventually contains records that seed the database tables and demonstrate its potential.

We must distinguish the UX of consuming "raw" OCA from products that consume OCA that are exposable to users. This doesn't mean the former isn't consumable by users, but it means we keep it coherent and on a certain level and leave the rest for the use case. In most (all?) cases, the latter should be the way to go, i.e., if someone struggles with "what to enter" here, this is because of a lack of proper tooling or solution within a product. If people digitally enter the data, the most known approach to follow is to create an appropriate form for it.

**Reply by carlyh-micb** [reply]
And yet examples are everywhere because they are useful.
Examples of data that could pass the rules contained in the schema are useful. They may not be business case specific, but as in the oca-spec people are smart enough to see an example of a valid data entry. Plus schema bundles can handle the swapping out of different examples, this is what OCA is built for - easy changing of features.

**Reply by blelump** [reply]
A reasonable example for a Bundle is its data record. Why propose a different construct for it?

**Reply by carlyh-micb** [reply]
Because many data records contain confidential information and many people are not allowed to look at them. Even the ones entering data.

**Reply by carlyh-micb** [reply]
BTW I need an example of a bank information to input, kindly supply your data record so that I can use it :-)

**Reply by blelump** [reply]
> Because many data records contain confidential information and many people are not allowed to look at them. Even the ones entering data.

You prepare such record the same way you'd prepare the example. It doesn't require the real data.

**Reply by carlyh-micb** [reply]
In a decentralized system it would be helpful to share the example in the schema, rather than each specific 'node' collaborating together but all coming up with their own examples. Examples shared via the schema can help ensure better data harmonization across a decentralized system.

A group will decide on a schema, create their examples and share this all via their schema, each implementing the systems that works best with their infrastructure but knowing that the shared example will help them harmonize data across their decentralized ecosystem.

**Reply by blelump** [reply]
The system architecture doesn't matter. People, when using it, just see the interface.  How it's organized under the hood is a detail they don't even see.

Example overlay is fully covered by the current nature of OCA and if a use case requires examples , it's up to this use case, because all the context is there to prepare such. However,  it at least questionable why a use case would need any.

The motivation in the RFC , to recap, `When collecting data from users, providing examples help users grasp the expected format or content of the data they need to input. This clarity reduces ambiguity and minimizes errors, enhancing overall data quality`, is a clear sign something is missing  in the approach to solve the problem. Viable options are: missing interface, process, or a tool that is incorporated into the process.
